### PR TITLE
[TS] LPS-127989 Inactive portal users cannot be exported to LDAP

### PR DESCRIPTION
**Pull request by istvansajtos**
[LPS-127989](https://issues.liferay.com/browse/LPS-127989)

Dear Reviewer,

My changes intend to enable exporting INACTIVE users to LDAP with the exception of anonymous users.

Related PTR ticket gives an explanation on why creating new method(s) for anonymous user check was necessary and how it was done. I'm not sure about the correct place of these methods, but I didn't find any util class in portal-security-ldap-impl.

Please let me know if you have any questions or concerns.

Thank you,
Istvan

### CODE REVIEW

File: modules/apps/asset/asset-categories-api/src/main/resources/content/Language.properties [hunk 1]
```diff
@@ -1,2 +1,3 @@
+categorization-configuration-name=Categorization
```
Comment (liferay-appsec) [inline modules/apps/asset/asset-categories-api/src/main/resources/content/Language.properties:1]: Check that added properties are either titles or sentences

File: modules/apps/captcha/captcha-api/src/main/resources/content/Language.properties [hunk 2]
```diff
@@ -1,3 +1,4 @@
+captcha-configuration-name=CAPTCHA
```
Comment (liferay-appsec) [inline modules/apps/captcha/captcha-api/src/main/resources/content/Language.properties:1]: Check that added properties are either titles or sentences

File: modules/apps/digital-signature/digital-signature-impl/src/main/java/com/liferay/digital/signature/internal/configuration/DigitalSignatureConfigurationDDMFormDeclaration.java [hunk 3]
```diff
@@ -0,0 +1,37 @@
+/**
```
Comment (liferay-appsec) [inline modules/apps/digital-signature/digital-signature-impl/src/main/java/com/liferay/digital/signature/internal/configuration/DigitalSignatureConfigurationDDMFormDeclaration.java:1]: When introducing a new class that implements an interface
        take into account:

* The class name should end with the interface name
* The package of the class matches the package of the interface



Refer to documentation:

File: modules/apps/document-library/document-library-item-selector-web/src/main/resources/content/Language.properties [hunk 4]
```diff
@@ -1,2 +1,2 @@
-dl-image-item-selector-view-name=Item Selector View
```
Comment (liferay-appsec) [inline modules/apps/document-library/document-library-item-selector-web/src/main/resources/content/Language.properties:1]: Check that added properties are either titles or sentences

File: modules/apps/password-policies-admin/password-policies-admin-web/src/main/resources/content/Language.properties [hunk 5]
```diff
@@ -50,6 +50,7 @@ password-changes=Password Changes
 password-expiration=Password Expiration
```
Comment (liferay-appsec) [inline modules/apps/password-policies-admin/password-policies-admin-web/src/main/resources/content/Language.properties:50]: Check that added properties are either titles or sentences

File: modules/apps/site/site-admin-web/src/main/resources/META-INF/resources/site_settings/analytics.jsp [hunk 6]
```diff
@@ -26,19 +26,35 @@ for (String analyticsType : analyticsTypes) {
 
 	<c:choose>
 		<c:when test='<%= StringUtil.equalsIgnoreCase(analyticsType, "google") %>'>
-			<aui:input helpMessage="set-the-google-analytics-id-that-is-used-for-this-set-of-pages" label="google-analytics-id" name="googleAnalyticsId" type="text" value='<%= PropertiesParamUtil.getString(groupTypeSettings, request, "googleAnalyticsId") %>' />
+			<aui:field-wrapper cssClass="form-group">
+				<aui:input label="google-analytics-id" name="googleAnalyticsId" type="text" value='<%= PropertiesParamUtil.getString(groupTypeSettings, request, "googleAnalyticsId") %>' />
```
Comment (liferay-appsec) [inline modules/apps/site/site-admin-web/src/main/resources/META-INF/resources/site_settings/analytics.jsp:30]: This line adds a form input. Please check:

* The input is validated on the processing action

* The backend is protected agains value forging



Refer to documentation:

File: modules/apps/site/site-admin-web/src/main/resources/META-INF/resources/site_settings/analytics.jsp [hunk 7]
```diff
@@ -26,19 +26,35 @@ for (String analyticsType : analyticsTypes) {
...
 
-			<aui:input helpMessage="set-the-google-analytics-create-custom-options-that-are-used-for-this-set-of-pages" label="google-analytics-create-custom-configuration" name="googleAnalyticsCreateCustomConfiguration" type="textarea" value='<%= PropertiesParamUtil.getString(groupTypeSettings, request, "googleAnalyticsCreateCustomConfiguration") %>' />
+				<span class="form-text"><liferay-ui:message key="set-the-google-analytics-id-that-is-used-for-this-set-of-pages" /></span>
+			</aui:field-wrapper>
 
-			<aui:input helpMessage="set-the-google-analytics-custom-options-that-are-used-for-this-set-of-pages" label="google-analytics-custom-configuration" name="googleAnalyticsCustomConfiguration" type="textarea" value='<%= PropertiesParamUtil.getString(groupTypeSettings, request, "googleAnalyticsCustomConfiguration") %>' />
+			<aui:field-wrapper cssClass="form-group">
+				<aui:input label="google-analytics-create-custom-configuration" name="googleAnalyticsCreateCustomConfiguration" type="textarea" value='<%= PropertiesParamUtil.getString(groupTypeSettings, request, "googleAnalyticsCreateCustomConfiguration") %>' />
```
Comment (liferay-appsec) [inline modules/apps/site/site-admin-web/src/main/resources/META-INF/resources/site_settings/analytics.jsp:36]: This line adds a form input. Please check:

* The input is validated on the processing action

* The backend is protected agains value forging



Refer to documentation:

File: modules/apps/site/site-admin-web/src/main/resources/META-INF/resources/site_settings/analytics.jsp [hunk 8]
```diff
@@ -26,19 +26,35 @@ for (String analyticsType : analyticsTypes) {
...
+
+				<span class="form-text"><liferay-ui:message key="set-the-google-analytics-create-custom-options-that-are-used-for-this-set-of-pages" /></span>
+			</aui:field-wrapper>
+
+			<aui:field-wrapper cssClass="form-group">
+				<aui:input label="google-analytics-custom-configuration" name="googleAnalyticsCustomConfiguration" type="textarea" value='<%= PropertiesParamUtil.getString(groupTypeSettings, request, "googleAnalyticsCustomConfiguration") %>' />
```
Comment (liferay-appsec) [inline modules/apps/site/site-admin-web/src/main/resources/META-INF/resources/site_settings/analytics.jsp:42]: This line adds a form input. Please check:

* The input is validated on the processing action

* The backend is protected agains value forging



Refer to documentation:

File: modules/apps/site/site-admin-web/src/main/resources/META-INF/resources/site_settings/analytics.jsp [hunk 9]
```diff
@@ -26,19 +26,35 @@ for (String analyticsType : analyticsTypes) {
...
+			</aui:field-wrapper>
 		</c:when>
 		<c:otherwise>
 
 			<%
 			String analyticsName = TextFormatter.format(analyticsType, TextFormatter.J);
 			%>
 
-			<aui:input helpMessage='<%= LanguageUtil.format(request, "set-the-script-for-x-that-is-used-for-this-set-of-pages", analyticsName, false) %>' label="<%= analyticsName %>" name="<%= Sites.ANALYTICS_PREFIX + analyticsType %>" type="textarea" value="<%= PropertiesParamUtil.getString(groupTypeSettings, request, Sites.ANALYTICS_PREFIX + analyticsType) %>" wrap="soft" />
+			<aui:field-wrapper cssClass="form-group">
```
Comment (liferay-appsec) [inline modules/apps/site/site-admin-web/src/main/resources/META-INF/resources/site_settings/analytics.jsp:53]: This line adds a form input. Please check:

* The input is validated on the processing action

* The backend is protected agains value forging



Refer to documentation:

File: modules/apps/site/site-admin-web/src/main/resources/META-INF/resources/site_settings/details.jsp [hunk 10]
```diff
@@ -105,89 +105,37 @@ if (parentGroupId != GroupConstants.DEFAULT_PARENT_GROUP_ID) {
...
-	List<Group> parentGroups = new ArrayList<Group>();
+	<div class="input-group mb-3">
+		<div class="input-group-item">
+			<input class="field form-control lfr-input-text" id="<portlet:namespace />parentSiteTitle" name="<portlet:namespace />parentSiteTitle" readonly="<%= true %>" value="<%= title %>" />
 
-	if (parentGroup != null) {
-		parentGroups.add(parentGroup);
-	}
-	%>
+			<aui:input name="parentGroupSearchContainerPrimaryKeys" type="hidden" value="<%= parentGroupId %>" />
```
Comment (liferay-appsec) [inline modules/apps/site/site-admin-web/src/main/resources/META-INF/resources/site_settings/details.jsp:124]: This line adds a form input. Please check:

* The input is validated on the processing action

* The backend is protected agains value forging



Refer to documentation:

File: modules/apps/site/site-admin-web/src/main/resources/META-INF/resources/site_settings/details.jsp [hunk 11]
```diff
@@ -200,31 +148,33 @@ if (parentGroupId != GroupConstants.DEFAULT_PARENT_GROUP_ID) {
...
+		manualMembership = GetterUtil.getBoolean(liveGroup.isManualMembership(), true);
+	}
+	%>
 
-					rowColumns.push(event.entityname);
-					rowColumns.push(event.grouptype);
-					rowColumns.push(
-						'<a class="modify-link" data-rowId="' +
-							event.entityid +
-							'" href="javascript:;"><%= UnicodeFormatter.toString(removeGroupIcon) %></a>'
```
Comment (liferay-appsec) [inline modules/apps/site/site-admin-web/src/main/resources/META-INF/resources/site_settings/details.jsp:222]: This line adds a form input. Please check:

* The input is validated on the processing action

* The backend is protected agains value forging



Refer to documentation:

File: modules/apps/site/site-admin-web/src/main/resources/META-INF/resources/site_settings/documents_and_media.jsp [hunk 12]
```diff
@@ -22,4 +22,8 @@ Group siteGroup = (Group)request.getAttribute("site.group");
 UnicodeProperties groupTypeSettings = (UnicodeProperties)request.getAttribute("site.groupTypeSettings");
 %>
 
-<aui:input helpMessage='<%= LanguageUtil.format(request, "directory-indexing-help", new Object[] {HtmlUtil.escape(siteGroup.getDescriptiveName(themeDisplay.getLocale())), themeDisplay.getPortalURL() + "/documents" + siteGroup.getFriendlyURL()}, false) %>' inlineLabel="right" label="enable-directory-indexing" labelCssClass="simple-toggle-switch" name="TypeSettingsProperties--directoryIndexingEnabled--" type="toggle-switch" value='<%= PropertiesParamUtil.getBoolean(groupTypeSettings, request, "directoryIndexingEnabled") %>' />
\ No newline at end of file
+<aui:field-wrapper cssClass="form-group">
+	<aui:input inlineLabel="right" label="enable-directory-indexing" labelCssClass="simple-toggle-switch" name="TypeSettingsProperties--directoryIndexingEnabled--" type="toggle-switch" value='<%= PropertiesParamUtil.getBoolean(groupTypeSettings, request, "directoryIndexingEnabled") %>' />
```
Comment (liferay-appsec) [inline modules/apps/site/site-admin-web/src/main/resources/META-INF/resources/site_settings/documents_and_media.jsp:26]: This line adds a form input. Please check:

* The input is validated on the processing action

* The backend is protected agains value forging



Refer to documentation:

File: modules/apps/site/site-admin-web/src/main/resources/META-INF/resources/site_settings/pages.jsp [hunk 13]
```diff
@@ -68,177 +68,181 @@ if (!LanguageUtil.isInheritLocales(siteGoup.getGroupId()) && !siteAdminConfigura
...
+				<c:choose>
+					<c:when test="<%= hasUnlinkLayoutSetPrototypePermission %>">
+						<div class="hide" id="<portlet:namespace />publicLayoutSetPrototypeIdOptions">
+							<c:if test="<%= disableLayoutSetPrototypeInput %>">
+								<div class="alert alert-info">
+									<liferay-ui:message key="you-cannot-enable-the-propagation-of-changes-because-you-modified-the-display-settings-of-this-site" />
+								</div>
+							</c:if>
+
+							<aui:input disabled="<%= disableLayoutSetPrototypeInput %>" helpMessage="enable-propagation-of-changes-from-the-site-template-help" inlineLabel="right" label="enable-propagation-of-changes-from-the-site-template" labelCssClass="simple-toggle-switch" name="publicLayoutSetPrototypeLinkEnabled" type="toggle-switch" value="<%= publicLayoutSetPrototypeLinkEnabled %>" />
```
Comment (liferay-appsec) [inline modules/apps/site/site-admin-web/src/main/resources/META-INF/resources/site_settings/pages.jsp:108]: This line adds a form input. Please check:

* The input is validated on the processing action

* The backend is protected agains value forging



Refer to documentation:

File: modules/apps/site/site-admin-web/src/main/resources/META-INF/resources/site_settings/pages.jsp [hunk 14]
```diff
@@ -68,177 +68,181 @@ if (!LanguageUtil.isInheritLocales(siteGoup.getGroupId()) && !siteAdminConfigura
...
+						</div>
+					</c:when>
+					<c:otherwise>
+						<aui:input name="publicLayoutSetPrototypeLinkEnabled" type="hidden" value="<%= true %>" />
```
Comment (liferay-appsec) [inline modules/apps/site/site-admin-web/src/main/resources/META-INF/resources/site_settings/pages.jsp:112]: This line adds a form input. Please check:

* The input is validated on the processing action

* The backend is protected agains value forging



Refer to documentation:

File: modules/apps/site/site-admin-web/src/main/resources/META-INF/resources/site_settings/pages.jsp [hunk 15]
```diff
@@ -68,177 +68,181 @@ if (!LanguageUtil.isInheritLocales(siteGoup.getGroupId()) && !siteAdminConfigura
...
+						<div class="alert alert-info">
+							<liferay-ui:message key="you-cannot-enable-the-propagation-of-changes-because-you-modified-the-display-settings-of-this-site" />
+						</div>
+					</c:if>
 
-			<%
-			for (LayoutSetPrototype curLayoutSetPrototype : layoutSetPrototypes) {
-				UnicodeProperties settingsProperties = curLayoutSetPrototype.getSettingsProperties();
-			%>
+					<aui:input disabled="<%= disableLayoutSetPrototypeInput %>" inlineLabel="right" label='<%= LanguageUtil.format(request, "enable-propagation-of-changes-from-the-site-template-x", HtmlUtil.escape(publicLayoutSetPrototype.getName(locale)), false) %>' labelCssClass="simple-toggle-switch" name="publicLayoutSetPrototypeLinkEnabled" type="toggle-switch" value="<%= publicLayoutSetPrototypeLinkEnabled %>" />
```
Comment (liferay-appsec) [inline modules/apps/site/site-admin-web/src/main/resources/META-INF/resources/site_settings/pages.jsp:135]: This line adds a form input. Please check:

* The input is validated on the processing action

* The backend is protected agains value forging



Refer to documentation:

File: modules/apps/site/site-admin-web/src/main/resources/META-INF/resources/site_settings/pages.jsp [hunk 16]
```diff
@@ -68,177 +68,181 @@ if (!LanguageUtil.isInheritLocales(siteGoup.getGroupId()) && !siteAdminConfigura
...
 
-		</aui:select>
+						<liferay-util:include page="/layout_set_merge_alert.jsp" servletContext="<%= application %>" />
+					</div>
+				</c:when>
+				<c:when test="<%= publicLayoutSetPrototype != null %>">
+					<liferay-ui:message arguments="<%= HtmlUtil.escape(publicLayoutSetPrototype.getName(locale)) %>" key="these-pages-are-linked-to-site-template-x" translateArguments="<%= false %>" />
 
-		<c:if test="<%= !siteGoup.isStaged() %>">
+					<aui:input name="publicLayoutSetPrototypeLinkEnabled" type="hidden" value="<%= publicLayoutSetPrototypeLinkEnabled %>" />
```
Comment (liferay-appsec) [inline modules/apps/site/site-admin-web/src/main/resources/META-INF/resources/site_settings/pages.jsp:152]: This line adds a form input. Please check:

* The input is validated on the processing action

* The backend is protected agains value forging



Refer to documentation:

File: modules/apps/site/site-admin-web/src/main/resources/META-INF/resources/site_settings/pages.jsp [hunk 17]
```diff
@@ -68,177 +68,181 @@ if (!LanguageUtil.isInheritLocales(siteGoup.getGroupId()) && !siteAdminConfigura
...
+				<c:choose>
+					<c:when test="<%= hasUnlinkLayoutSetPrototypePermission %>">
+						<div class="hide" id="<portlet:namespace />privateLayoutSetPrototypeIdOptions">
+							<c:if test="<%= disableLayoutSetPrototypeInput %>">
+								<div class="alert alert-info">
+									<liferay-ui:message key="you-cannot-enable-the-propagation-of-changes-because-you-modified-the-display-settings-of-this-site" />
+								</div>
+							</c:if>
+
+							<aui:input disabled="<%= disableLayoutSetPrototypeInput %>" helpMessage="enable-propagation-of-changes-from-the-site-template-help" inlineLabel="right" label="enable-propagation-of-changes-from-the-site-template" labelCssClass="simple-toggle-switch" name="privateLayoutSetPrototypeLinkEnabled" type="toggle-switch" value="<%= privateLayoutSetPrototypeLinkEnabled %>" />
```
Comment (liferay-appsec) [inline modules/apps/site/site-admin-web/src/main/resources/META-INF/resources/site_settings/pages.jsp:196]: This line adds a form input. Please check:

* The input is validated on the processing action

* The backend is protected agains value forging



Refer to documentation:

File: modules/apps/site/site-admin-web/src/main/resources/META-INF/resources/site_settings/pages.jsp [hunk 18]
```diff
@@ -68,177 +68,181 @@ if (!LanguageUtil.isInheritLocales(siteGoup.getGroupId()) && !siteAdminConfigura
...
+						</div>
+					</c:when>
+					<c:otherwise>
+						<aui:input name="privateLayoutSetPrototypeLinkEnabled" type="hidden" value="<%= true %>" />
```
Comment (liferay-appsec) [inline modules/apps/site/site-admin-web/src/main/resources/META-INF/resources/site_settings/pages.jsp:200]: This line adds a form input. Please check:

* The input is validated on the processing action

* The backend is protected agains value forging



Refer to documentation:

File: modules/apps/site/site-admin-web/src/main/resources/META-INF/resources/site_settings/pages.jsp [hunk 19]
```diff
@@ -68,177 +68,181 @@ if (!LanguageUtil.isInheritLocales(siteGoup.getGroupId()) && !siteAdminConfigura
...
+			<c:choose>
+				<c:when test="<%= (privateLayoutSetPrototype != null) && !siteGoup.isStaged() && hasUnlinkLayoutSetPrototypePermission %>">
+					<c:if test="<%= disableLayoutSetPrototypeInput %>">
+						<div class="alert alert-info">
+							<liferay-ui:message key="you-cannot-enable-the-propagation-of-changes-because-you-modified-the-display-settings-of-this-site" />
+						</div>
+					</c:if>
 
-				<div class="<%= privateLayoutSetPrototypeLinkEnabled ? "" : "hide" %>" id="<portlet:namespace />privateLayoutSetPrototypeMergeAlert">
+					<aui:input disabled="<%= disableLayoutSetPrototypeInput %>" inlineLabel="right" label='<%= LanguageUtil.format(request, "enable-propagation-of-changes-from-the-site-template-x", HtmlUtil.escape(privateLayoutSetPrototype.getName(locale)), false) %>' labelCssClass="simple-toggle-switch" name="privateLayoutSetPrototypeLinkEnabled" type="toggle-switch" value="<%= privateLayoutSetPrototypeLinkEnabled %>" />
```
Comment (liferay-appsec) [inline modules/apps/site/site-admin-web/src/main/resources/META-INF/resources/site_settings/pages.jsp:223]: This line adds a form input. Please check:

* The input is validated on the processing action

* The backend is protected agains value forging



Refer to documentation:

File: modules/apps/site/site-admin-web/src/main/resources/META-INF/resources/site_settings/pages.jsp [hunk 20]
```diff
@@ -68,177 +68,181 @@ if (!LanguageUtil.isInheritLocales(siteGoup.getGroupId()) && !siteAdminConfigura
...
+						request.setAttribute("edit_layout_set_prototype.jsp-redirect", currentURL);
+						%>
+
+						<liferay-util:include page="/layout_set_merge_alert.jsp" servletContext="<%= application %>" />
+					</div>
+				</c:when>
+				<c:when test="<%= privateLayoutSetPrototype != null %>">
+					<liferay-ui:message arguments="<%= HtmlUtil.escape(privateLayoutSetPrototype.getName(locale)) %>" key="these-pages-are-linked-to-site-template-x" translateArguments="<%= false %>" />
+
+					<aui:input name="privateLayoutSetPrototypeLinkEnabled" type="hidden" value="<%= privateLayoutSetPrototypeLinkEnabled %>" />
```
Comment (liferay-appsec) [inline modules/apps/site/site-admin-web/src/main/resources/META-INF/resources/site_settings/pages.jsp:240]: This line adds a form input. Please check:

* The input is validated on the processing action

* The backend is protected agains value forging



Refer to documentation:

File: modules/apps/site/site-admin-web/src/main/resources/META-INF/resources/site_settings/recycle_bin.jsp [hunk 21]
```diff
@@ -26,11 +26,13 @@ int trashEntriesMaxAge = PropertiesParamUtil.getInteger(groupTypeSettings, reque
 
 <aui:input id="trashEnabled" inlineLabel="right" label="enable-recycle-bin" labelCssClass="simple-toggle-switch" name="trashEnabled" type="toggle-switch" value="<%= groupTrashEnabled %>" />
 
-<div class="trash-entries-max-age">
-	<aui:input disabled="<%= !groupTrashEnabled %>" helpMessage="trash-entries-max-age-help" label="trash-entries-max-age" name="trashEntriesMaxAge" type="text" value="<%= ((trashEntriesMaxAge % 1) == 0) ? GetterUtil.getInteger(trashEntriesMaxAge) : String.valueOf(trashEntriesMaxAge) %>">
+<aui:field-wrapper cssClass="form-group">
+	<aui:input disabled="<%= !groupTrashEnabled %>" label="trash-entries-max-age" name="trashEntriesMaxAge" type="text" value="<%= ((trashEntriesMaxAge % 1) == 0) ? GetterUtil.getInteger(trashEntriesMaxAge) : String.valueOf(trashEntriesMaxAge) %>">
```
Comment (liferay-appsec) [inline modules/apps/site/site-admin-web/src/main/resources/META-INF/resources/site_settings/recycle_bin.jsp:30]: This line adds a form input. Please check:

* The input is validated on the processing action

* The backend is protected agains value forging



Refer to documentation:

File: modules/apps/site/site-admin-web/src/main/resources/META-INF/resources/site_settings/site_url.jsp [hunk 22]
```diff
@@ -152,7 +153,7 @@ if (privateVirtualHostnames.isEmpty()) {
 				cssClass="lfr-form-row"
 			>
 				<clay:row>
-					<aui:input inlineField="<%= true %>" label="public-pages" maxlength="200" name="publicVirtualHostname[]" placeholder="virtual-host" type="text" value="<%= virtualHostname %>" wrapperCssClass="col-sm-6" />
```
Comment (liferay-appsec) [inline modules/apps/site/site-admin-web/src/main/resources/META-INF/resources/site_settings/site_url.jsp:155]: This line adds a form input. Please check:

* The input is validated on the processing action

* The backend is protected agains value forging



Refer to documentation:

File: modules/apps/site/site-admin-web/src/main/resources/META-INF/resources/site_settings/site_url.jsp [hunk 23]
```diff
@@ -191,7 +193,7 @@ if (privateVirtualHostnames.isEmpty()) {
 				cssClass="lfr-form-row"
 			>
```
Comment (liferay-appsec) [inline modules/apps/site/site-admin-web/src/main/resources/META-INF/resources/site_settings/site_url.jsp:194]: This line adds a form input. Please check:

* The input is validated on the processing action

* The backend is protected agains value forging



Refer to documentation:

File: modules/apps/site/site-admin-web/src/main/resources/content/Language.properties [hunk 24]
```diff
@@ -4,6 +4,7 @@ basic-information=Basic Information
 category.community-tools=Community Tools
```
Comment (liferay-appsec) [inline modules/apps/site/site-admin-web/src/main/resources/content/Language.properties:4]: Check that added properties are either titles or sentences

File: modules/apps/translation/translation-google-cloud-translator/src/main/java/com/liferay/translation/google/cloud/translator/internal/configuration/persistence/listener/GoogleCloudTranslatorConfigurationModelListener.java [hunk 25]
```diff
@@ -0,0 +1,83 @@
+/**
```
Comment (liferay-appsec) [inline modules/apps/translation/translation-google-cloud-translator/src/main/java/com/liferay/translation/google/cloud/translator/internal/configuration/persistence/listener/GoogleCloudTranslatorConfigurationModelListener.java:1]: When introducing a new class that implements an interface
        take into account:

* The class name should end with the interface name
* The package of the class matches the package of the interface



Refer to documentation:

File: modules/apps/translation/translation-google-cloud-translator/src/main/resources/content/Language.properties [hunk 26]
```diff
@@ -2,4 +2,5 @@ category.translation=Translation
 enabled-description=Enable or disable the integration with Google Cloud Translation API for automatic translations.
```
Comment (liferay-appsec) [inline modules/apps/translation/translation-google-cloud-translator/src/main/resources/content/Language.properties:2]: Check that added properties are either titles or sentences

File: modules/dxp/apps/portal-search-tuning/portal-search-tuning-synonyms-web/src/main/resources/content/Language.properties [hunk 27]
```diff
@@ -1,9 +1,9 @@
-broaden-the-scope-of-search-by-treating-terms-equally-using-synonyms=Broaden the scope of search by treating terms equally using synonyms.
```
Comment (liferay-appsec) [inline modules/dxp/apps/portal-search-tuning/portal-search-tuning-synonyms-web/src/main/resources/content/Language.properties:1]: Check that added properties are either titles or sentences

**Review by liferay-appsec — COMMENTED** [review]
This is an automatic review suggestion

**Comment by istvansajtos** [comment]
ci:help

**Comment by istvansajtos** [comment]
ci:test:sf:norebase

**Comment by istvansajtos** [comment]
ci:test:relevant:norebase

**Comment by istvansajtos** [comment]
ci:test:ldap:norebase

**Comment by arthurchan35** [comment]
See: https://github.com/liferay-appsec/liferay-portal/pull/368

**Pull request closed without merging**